### PR TITLE
.NET: Restore Packages command should work with .NET Core .csproj projects

### LINKED ISSUE #1175: 'dotnet: Restore Packages' VS Code command does not work
## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (1.0.0-rc3-004530)

Product Information:
 Version:            1.0.0-rc3-004530
 Commit SHA-1 hash:  0de3338607

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.0-rc3-004530
```
.Net core installer downloaded from: https://github.com/dotnet/core/blob/master/release-notes/rc3-download.md

VS Code version: 1.9.0 64-bit (on Windows 10 Pro 1607)

C# Extension version: 1.6.2, 1.7.0-beta2

## Steps to reproduce
1. Extract attached project:
[mvc5.zip](https://github.com/OmniSharp/omnisharp-vscode/files/750302/mvc5.zip)
Project was created by running `dotnet new -t web1.1`
2. Open project's folder in Visual Studio Code

## Expected  behavior
1. On load, a package restore is done and user can work with the project

## Actual behavior
Restore is not done, no namespaces are resolved. OmniSharp produces the following output:
```
Starting OmniSharp server at 3.02.2017, 12:33:45
    Target: c:\Users\milosz.kosobucki\Documents\asptest\mvc5

OmniSharp server started
    Path: C:\Users\milosz.kosobucki\.vscode\extensions\ms-vscode.csharp-1.6.2\bin\omnisharp\OmniSharp.exe
    PID: 32

OmniSharp: -s c:\Users\milosz.kosobucki\Documents\asptest\mvc5 --hostPID 9536 --stdio DotNet:enablePackageRestore=false --encoding utf-8 formattingOptions:useTabs=false formattingOptions:tabSize=4 formattingOptions:indentationSize=4
[INFORMATION:OmniSharp.Startup] Omnisharp server running using Stdio at location 'c:\Users\milosz.kosobucki\Documents\asptest\mvc5' on host 9536.
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Initializing in c:\Users\milosz.kosobucki\Documents\asptest\mvc5
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Auto package restore: False
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Update workspace context
[INFORMATION:OmniSharp.DotNet.DotNetProjectSystem] Resolving projects references
[INFORMATION:OmniSharp#MSBuild] MSBUILD_EXE_PATH environment variable set to C:\Users\milosz.kosobucki\.vscode\extensions\ms-vscode.csharp-1.6.2\bin\omnisharp\msbuild\MSBuild.exe
[INFORMATION:OmniSharp#MSBuild] MSBuildExtensionsPath environment variable set to C:\Users\milosz.kosobucki\.vscode\extensions\ms-vscode.csharp-1.6.2\bin\omnisharp\msbuild
[INFORMATION:OmniSharp#MSBuild] MSBuildSDKsPath environment variable set to C:\Users\milosz.kosobucki\.vscode\extensions\ms-vscode.csharp-1.6.2\bin\omnisharp\msbuild\Sdks
[INFORMATION:OmniSharp#MSBuild] No solution files found in 'c:\Users\milosz.kosobucki\Documents\asptest\mvc5'
[INFORMATION:OmniSharp#MSBuild] Add project: c:\Users\milosz.kosobucki\Documents\asptest\mvc5\mvc5.csproj
[INFORMATION:OmniSharp#MSBuild] Update project: mvc5
[INFORMATION:OmniSharp.Script.ScriptProjectSystem] Detecting CSX files in 'c:\Users\milosz.kosobucki\Documents\asptest\mvc5'.
[INFORMATION:OmniSharp.Script.ScriptProjectSystem] Could not find any CSX files
[INFORMATION:OmniSharp.Startup] Configuration finished.
```

Manually running Ctrl+Shift+p and typing `dotnet restore` results in:

image

### PR DESCRIPTION
**Pull request by DustinCampbell**
fixes #1175

### CODE REVIEW

File: src/omnisharp/protocol.ts [hunk 1]
```diff
@@ -509,4 +511,33 @@ export namespace V2 {
         Failure: string;
         Pass: boolean;
     }
+}
+
+export function findNetCoreAppTargetFramework(project: MSBuildProject): TargetFramework {
+    return project.TargetFrameworks.find(tf => tf.ShortName.startsWith('netcoreapp'));
```
Comment (filipw) [inline src/omnisharp/protocol.ts]: should also be `netstandard`?

Comment (DustinCampbell) [inline src/omnisharp/protocol.ts]: *grumble* yes.

Comment (DustinCampbell) [inline src/omnisharp/protocol.ts]: :smile:

**Comment by DustinCampbell** [comment]
Grr... That 'netstandard' "fix" I made will break other things. Fixing again and force-pushing.

**Comment by filipw** [comment]
great, looking forward to this

**Pull request merged**